File: old/apps/backend/src/modules/addon-env/addon-env.entity.ts
import { Entity, ManyToOne, PrimaryKey, Property } from '@mikro-orm/core';
import { v4 } from 'uuid';

import { Addon } from '../addon/addon.entity';

@Entity({ tableName: 'addon_envs' })
export class AddonEnv {
  @PrimaryKey()
  id: string = v4();

  @Property()
  key: string;

  @Property()
  value: string;

  @ManyToOne(() => Addon)
  addon: Addon;

  @Property({ onCreate: () => new Date() })
  creationDate: Date;

  @Property({ onUpdate: () => new Date() })
  updateTime: Date;
}

File: backend/src/main/java/com/esmo/empaas/entities/AddonEnvEntity.java
package com.esmo.empaas.entities;

import java.util.Date;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "addon_envs")
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class AddonEnvEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(name = "key")
    private String key;

    @Column(name = "value")
    private String value;

    @Column(name = "optional")
    private boolean optional;

    @Column(name = "explanation", columnDefinition = "TEXT")
    private String explanation;

    @ManyToOne
    @JoinColumn(name = "addon_id")
    private AddonEntity addon;

    @CreationTimestamp
    @Column(name = "creation_date")
    private Date creationDate;

    @UpdateTimestamp
    @Column(name = "modification_date")
    private Date modificationDate;
}